### PR TITLE
Add support for passing a factory method in module config directly ra…

### PR DESCRIPTION
**Pull request by djMax**
…ther than meddelware resolving

### CODE REVIEW

**Comment by shaunwarman** [comment]
Hey @djMax ! Apologies for the dumb question, but could you elaborate on the use case? Also, would using shortstop to reference a module (e.g. `"module": "./lib/mymodule"`) not do the same to take advantage of `arguments`?

File: index.js [hunk 1]
```diff
@@ -79,23 +85,41 @@ function resolveImpl(root, config) {
...
+
+        // If modulePath was not resolved lookup with config.name for meaningful error message.
+        module = require(modulePath || config.name);
+
+        // First, look for a factory method in the config
+        if (config.method) {
+            if (module[config.method] && thing.isFunction(module[config.method])) {
+                // Straight named export
+                factory = module[config.method];
+            } else if (module.default && this.isObject(module.default)) {
```
Comment (shaunwarman) [inline index.js]: `thing.isObject`?

**Comment by djMax** [comment]
The use case here is that you can't actually use shortstop to resolve a function - you have to use meddleware's resolution logic. Right now, if "module" points to a function, meddleware will throw because it's not a string (right?).

There are a variety of reasons I need this (async handling being one) including transpilation - where the location of that file is not a simple relative path in relation to the config. Shortstop already handles this mess very well, but I can't use it in meddleware w/o this change.

**Comment by djMax** [comment]
What's the disposition here? If it's not going to be merged we'll just fork private and move on.

**Comment by tlivings** [comment]
@jasisk should weigh in here.

**Comment by jasisk** [comment]
Addressing this with the following constraints:
1. Pretending you're not my boss (gulp!)
2. Pretending I don't know the context

My concern with this PR is that it promotes coupling over separation. With the current API, each middleware needs to be defined as a separate factory. Since it's a factory, any init time variables can be passed in as arguments to influence what middleware is _actually_ consumed in the continuation. That should address nearly all use-cases.

Why do you need to pass in a factory function instead of requiring a module?

**Comment by djMax** [comment]
I don't NEED meddleware to do this. But I think meddleware is taking an opinionated approach to the resolution of a name to code. In our case, partly for async, partly for dynamic handling, WE want to decide how that module configuration is resolved to code. Thus the factory method (because reusing name would be bad).

**Pull request closed without merging**